common/src/main/java/org/corfudb/common/metrics/micrometer/registries/dropwizard/DropwizardMeterRegistryWithSlidingTimeWindow.java

```diff
@@ -0,0 +1,63 @@
+package org.corfudb.common.metrics.micrometer.registries.dropwizard;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.SlidingTimeWindowArrayReservoir;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
+import io.micrometer.core.instrument.distribution.pause.PauseDetector;
+import io.micrometer.core.instrument.dropwizard.DropwizardConfig;
+import io.micrometer.core.instrument.dropwizard.DropwizardMeterRegistry;
+import io.micrometer.core.instrument.util.HierarchicalNameMapper;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class overrides the timer and histogram of DropwizardMeterRegistry and adds a
+ * SlidingWindowReservoirArray. This allows for the reporting of measurements made within the last
+ * n TimeUnits. This is the default behavior in other micrometer registries.
+ */
+public class DropwizardMeterRegistryWithSlidingTimeWindow extends DropwizardMeterRegistry {
```

Comment:
Seems to be unused. Will it be used in the next PR?

Reply:
It's used by the clients on their side within the RegistryProvider implementation.